pkg/storage/image.go

```diff
@@ -555,6 +556,13 @@ func (svc *imageService) ResolveNames(imageName string) ([]string, error) {
 	}
 	domain, remainder := splitDockerDomain(imageName)
 	if domain != "" {
+		registry, err := sysregistriesv2.FindRegistry(nil, domain)
+		if err != nil {
+			return nil, err
+		}
+		if registry != nil && registry.Blocked {
+			return nil, fmt.Errorf("cannot resolve names from registry %q because it's blocked", domain)
+		}
 		// this means the image is already fully qualified
```

Comment:
This comment logically belongs at top of this `if domain != ""` branch, above the added code.